### PR TITLE
[slaunch] Add preliminary support for loading a custom uncompressed ARGB image format.

### PR DESCRIPTION
**Pull request by DarjanKrijan**
Comparison video as announced:
https://www.youtube.com/watch?v=JbJaZhiyz-0

Probably now limited by rendering other stuff (bg image -> could also use ARGB now, also backdrop, etc.).

PS quick note:
Something is broken with your recent commit [4b4838da03745c73b29c524fd693f1d66f3b26bb](https://github.com/aldostools/webMAN-MOD/commit/4b4838da03745c73b29c524fd693f1d66f3b26bb)
Crashes instantly upon opening even with clean/removed wmtmp.

### CODE REVIEW

**Comment by aldostools** [comment]
Thank you for your new code. It looks very fast in your test video.

In regards to the broken code, I couldn't reproduce the bug. I double checked the commit and it looks fine.
I found a bug in a commented code, so it shouldn't be the issue. I guess your issue could be a NULL string being used in a strstr() or in another function. I submitted some changes that you could test if that fixes the bug.

**Comment by DarjanKrijan** [comment]
Think I had -O3 in my Makefile, could have produced buggy code or ended up using too much memory.
My bad, works with -Os.

**Comment by aldostools** [comment]
Nice to know it was a bad compilation. I have a new commit that let browse the pages very fast.

I think this improvement now makes the ARGB irrelevant. Anyway the support is still there.

**Comment by DarjanKrijan** [comment]
Nice feature, I like it.
It slowed down the cover drawing a bit as I guess it waits a frame for checking user input. Still, preferring this.

I'd still leave ARGB in for now if it doesn't cause any issues (code size maybe), it should still be much quicker drawing the covers.
Experiment around with it, could use it for the assets in `wm_icons` as well, it's basically a cached format for directly drawing to screen.

By the way, I'm gonna try to yank out this highly cell optimized `memcpy` utilizing prefetching and unrolling from glibc for using it in WMM:
https://github.com/bminor/glibc/blob/master/sysdeps/powerpc/powerpc64/cell/memcpy.S
Original patch from IBM: https://sourceware.org/legacy-ml/libc-alpha/2010-01/msg00029.html

Basically all this line drawing to the frame buffer can/should be replaced by a (that) SIMD optimized `memcpy` function instead of manually looping through trying to act smart with typecasting to `uint64_t`, `vec_uint4`, etc., this should save code size as well if done everywhere. Same for other programs/plugins in this suite. Will see what I can do.

There seems to be one simpler optimized `memcpy` in the cell SDK as well, one patch message tells it's buggy though, not sure it got fixed at some point. Just FYI.

**Comment by aldostools** [comment]
Thank you for your feedback. The optimized memcpy sounds great.

I implemented a new memset32 that theorically is 4X faster than memset.
Not sure if it's really needed after the commit c1727e699dfbdd18dbfc1a85d05b8e5466b46cb6

The frame rate in sLaunch 1.15a now is so fast that I had to add a 10ms delay in pad_read. The issue was that get_vsh_pad_obj() was scanning the VSH memory everytime MyPadGetData was called (on every frame). Now io_pad_object_address is cached after find it the first time.

**Comment by DarjanKrijan** [comment]
Nvm about the remark that covers load slower, it's quick af again after I updated to latest WMM!

**Comment by DarjanKrijan** [comment]
Edit:
Moved here: https://github.com/aldostools/webMAN-MOD/commit/4b4838da03745c73b29c524fd693f1d66f3b26bb#commitcomment-67790315
